### PR TITLE
Add the commit revision to binary/reports/docker

### PR DESCRIPTION
**Pull request by Munsio**

### CODE REVIEW

**Comment by Munsio** [comment]
@ruiAzevedo19 please review

Testing the revision with the binary and a golang test itself would mean to run system tests which we currently do not have. Therefore the normal test with the check for `development` is the only one.

The docker image one can be reviewed here: https://github.com/symflower/eval-dev-quality/actions/runs/9875489063/job/27272302420?pr=255#step:6:41
Please note that the Branch Commit hash and the PR commit hash are different (unfortunately) But  we are building on Branches now too which you can check with: https://github.com/symflower/eval-dev-quality/actions/runs/9875489063/job/27272302420#step:6:41 This one has to be the same as the latest commit in the branch itself.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -34,3 +34,5 @@ jobs:
           platforms: linux/amd64
           tags: ${{ steps.meta.outputs.tags }}
           push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```
Comment (ruiAzevedo19) [inline .github/workflows/docker.yml:38]: Can you explain me this please?

File: .github/workflows/branch.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Branch CI
+
+on:
+  push:
+    branches-ignore:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```
Comment (ruiAzevedo19) [inline .github/workflows/branch.yml:10]: The group here creates an ID for the current CI run right?

File: cmd/eval-dev-quality/cmd/version.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// SetLogger sets the logger of the command.
+func (command *Version) SetLogger(logger *log.Logger) {
+	command.logger = logger
+}
+
+// Execute executes the command.
+func (command *Version) Execute(args []string) (err error) {
+	revision := evaluate.Revision
+	if evaluate.Revision == "" {
```
Comment (ruiAzevedo19) [inline cmd/eval-dev-quality/cmd/version.go:24]: Maybe we can use `if revision == "" { ... }`

File: .github/workflows/branch.yml (hunk 2 above)
Comment (Munsio) [reply .github/workflows/branch.yml:10]: Yes. With this line we tell Github Actions to only have the latest CI running and abort the previous one. This is especially helpful if you push multiple times in a short time.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (Munsio) [reply .github/workflows/docker.yml:38]: This is to tell the Docker build step to push/fetch data from the Github Cache so building the container is faster on subsequent runs.

Comment (ruiAzevedo19) [reply .github/workflows/docker.yml:38]: Thanks

**Comment by ruiAzevedo19** [comment]
@bauersimon Please, do a second review

File: .github/workflows/branch.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  docker:
+    name: 'Run docker image building'
+    uses: ./.github/workflows/docker.yml
```
Comment (bauersimon) [inline .github/workflows/branch.yml:15]: what is with the `ci.yml` workflow? it also builds the docker image right? do we have it twice now?

**Review by bauersimon — COMMENTED** [review]
awesome - tried it locally. works great

File: .github/workflows/branch.yml (hunk 4 above)
Comment (bauersimon) [reply .github/workflows/branch.yml:15]: maybe we should drop the PR building then since we do the branch anyways?

Comment (bauersimon) [reply .github/workflows/branch.yml:15]: well... we can always remove that afterwards... better to have too many than too less ^^

**Review by bauersimon — APPROVED** [review]

**Pull request merged**